iceoryx2-cxx/src/node_name.cpp

```diff
@@ -79,23 +84,23 @@ void NodeName::drop() noexcept {
     }
 }
 
-auto NodeName::create(const char* value) -> iox2::legacy::expected<NodeName, bb::SemanticStringError> {
+auto NodeName::create(const char* value) -> iox2::container::Expected<NodeName, bb::SemanticStringError> {
     return NodeName::create_impl(value, strnlen(value, IOX2_NODE_NAME_LENGTH + 1));
 }
 
 auto NodeName::create_impl(const char* value, size_t value_len)
-    -> iox2::legacy::expected<NodeName, bb::SemanticStringError> {
+    -> iox2::container::Expected<NodeName, bb::SemanticStringError> {
     if (value_len > IOX2_NODE_NAME_LENGTH) {
-        return iox2::legacy::err(bb::SemanticStringError::ExceedsMaximumLength);
+        return { container::unexpect, bb::SemanticStringError::ExceedsMaximumLength };
     }
 
     iox2_node_name_h handle {};
     const auto ret_val = iox2_node_name_new(nullptr, value, value_len, &handle);
     if (ret_val == IOX2_OK) {
-        return iox2::legacy::ok(NodeName { handle });
+        return { container::in_place, NodeName { handle } };
     }
 
-    return iox2::legacy::err(iox2::bb::into<bb::SemanticStringError>(ret_val));
+    return { container::unexpect, iox2::bb::into<bb::SemanticStringError>(ret_val) };
```

Comment:
@ComicSansMS @elfenpiff @FerdinandSpitzschnueffler these helper structs are unfortunately not possible anymore, since the conversion methods would have to be part of the respective `expected` implementation.

`ok` and `err` felt a little bit more obvious for the happy and sad path. Maybe it is just a matter of getting used to `in_place` and `unexpect`. An option would be to introduce `IOX2_OK` and `IOX2_ERR` macros but unfortunately, there is already a `IOX2_OK` define from the C bindings.

What is your opinion?